sonaranalyzer-dotnet/src/SonarAnalyzer.CFG/Helpers/StringExtensions.cs

```diff
@@ -23,49 +23,60 @@
 
 namespace SonarAnalyzer.CFG.Helpers
 {
-    internal static class StringExtensions
+    public static class StringExtensions
     {
         /// <summary>
         /// Splits the input string to the list of words.
         ///
-        /// Letters and consecutive capital letters are ignored.
+        /// Sequence of upper case letters is considered as single word.
+        ///
         /// For example:
         /// thisIsAName => this is a name
-        /// ThisIsIt => this is it
+        /// ThisIsSMTPName => this is smtp name
         /// bin2hex => bin hex
-        /// HTML => h t m l
-        /// PEHeader => p e header
+        /// HTML => html
+        /// SOME_value => some value
+        /// PEHeader => pe header
         /// </summary>
         /// <param name="name">A string containing words.</param>
         /// <returns>A list of words (all lowercase) contained in the string.</returns>
         public static IEnumerable<string> SplitCamelCaseToWords(this string name)
```

Comment:
Although we cannot do anything regarding coverage, I think we could improve the duplication.
I looked at the code, and this method is only used [in a single place](https://github.com/SonarSource/sonar-dotnet/blob/1946c1b77e124446a38f301bb7fd10f67a0f0fce/sonaranalyzer-dotnet/src/SonarAnalyzer.CFG/ControlFlowGraph/CfgSerializer.cs#L128) in the CFG assembly (it is not used either in the security assembly, as it is present there as another duplicated method...).

I'd suggest to remove the implementation from CFG dll and use a simplified version inside the `WriteNode` method given that the context in which it is used is limited.

WDYT ?

Reply:
I don't like the duplication, but I replaced it anyway. It will be easier to decouple the helpers layer in the future if we keep it as close as possible.

All helpers in CFG are duplicates by-design, for now.

Reply:
Ok. Let's keept it like this for now, and indeed see if/when we will have to decouple some logic between these projects